app/routes/home/admin.js

```diff
@@ -0,0 +1,42 @@
+import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+
+export default class AdminRoute extends Route {
+  @service session;
+  @service currentUser;
+  @service router;
+
+  beforeModel(transition) {
+    // First check if user is authenticated
+    if (!this.session.isAuthenticated) {
+      this.session.requireAuthentication(transition, 'home.login');
+      return;
+    }
+
+    // Load current user if not already loaded
+    return this.currentUser.load();
+  }
+
+  afterModel() {
+    // Check if user has admin role
+    if (!this.currentUser.user || !this.currentUser.user.roles.includes('admin')) {
+      // Redirect non-admin users to home page
+      this.router.transitionTo('home.index');
+    }
+  }
+
+  model() {
+    // For now, return fake data
+    // Eventually this will come from /api/admin_stats.json
+    return {
```

Comment:
The fruit counts can actually be retreived from the chat.

take a simliar approach that app/components/fruit-tip/fruit.ts does and inject the chat service into the admin controller, calling this.chat.getFruitCount there